src/EnergyPlus/PlantLoopHeatPumpEIR.cc

```diff
@@ -1486,6 +1492,15 @@ void EIRPlantLoopHeatPump::processInputForEIRPLHP(EnergyPlusData &state)
                                                                                      condenserNodeConnectionType_Inlet,
                                                                                      NodeInputManager::CompFluidStream::Secondary,
                                                                                      DataLoopNode::ObjectIsNotParent);
+                if (condenserNodeType == DataLoopNode::NodeFluidType::Air) {
+                    if (!OutAirNodeManager::CheckOutAirNodeNumber(state, thisPLHP.sourceSideNodes.inlet)) {
+                        ShowSevereError(
+                            state,
+                            format(
+                                "Air Source PlantLoop:HeatPump: {} inlet node: {} is not an OutdoorAir:Node.", thisPLHP.name, sourceSideInletNodeName));
+                        ShowContinueError(state, "Confirm that this is the intended source for the outdoor air stream.");
+                    }
+                }
```

Comment:
Although we don't enforce it in the code in any way, moving forward, it would be nice if we only issue severe errors that will ultimately result in a fatal.  If the simulation is able to continue to run without stopping, make it a warning.  I think it's worth changing that before merging this, but I'm open to discussion.

Reply:
I'm open to changing the error, I think it makes sense to discuss whether it should be a fatal error, or simply a warning.

If the inlet node is not an outdoor air node, then the inlet sees 0C for all timesteps.  Perhaps we could extend the check beyond Outdoor air nodes?  In this case, we issue a warning if the inlet node is something other than an Outdoor Air Node, and issue a Severe (followed by Fatal) if the inlet node is not found anywhere else in the model.

Reply:
So, the root problem here is the [calls to `getOnlySingleNode`](https://github.com/NREL/EnergyPlus/blob/dddf02031ab8bbde658a486d41ce0b1c4553649e/src/EnergyPlus/PlantLoopHeatPumpEIR.cc#L2803-L2820) for the air-cooled condenser case. If these were `DataLoopNode::ConnectionType::Inlet and Outlet` instead of `DataLoopNode::ConnectionType::OutsideAir` the regular node checking would warn about an inlet that's not connected to anything, and it would fatal (I'm 99% certain) if the same node is used a an inlet and outlet on the same equipment.

You may still want to issue a warning if the inlet is connected to something other than outdoor air, or maybe not.

The next question then is whether the conditions are properly set on the air-cooled condenser outlet node? If not, that may be why they were set to `OutsideAir` instead of `inlet/outlet`. Do we have an air volume flow rate for the air cooled condenser?

Reply:
Source side outlet condition is set in doPhysics. I agree that these should be inlet/outlet. P.S. there are some `// TRANE` comments in simulate (around line 100) that can be removed.

Reply:
@mjwitte - what do you mean "do we have an air volume flow rate for the air cooled condenser?"  

EnergyPlus certainly calculates a source side mass flow rate when the heat pump source is Air:

`Output:Variable,*,Heat Pump Source Side Mass Flow Rate,hourly; !- HVAC Average [kg/s]`

Reply:
Ok. So, it should be valid to set that as generic inlet and outlet nodes and let the user connect them to whatever they want. 

I just wasn't sure about this equipment, because some air-cooled equipment (like DX coils) doesn't calculate any condenser flow rate or condenser outlet conditions, it just uses the condenser inlet conditions to feed into the performance curves.

Reply:
I changed:

```
                    condenserNodeConnectionType_Inlet = DataLoopNode::ConnectionType::OutsideAir;
                    condenserNodeConnectionType_Outlet = DataLoopNode::ConnectionType::OutsideAir;
```

to:

```
                    condenserNodeConnectionType_Inlet = DataLoopNode::ConnectionType::Inlet;
                    condenserNodeConnectionType_Outlet = DataLoopNode::ConnectionType::Outlet;
```

Standard node checking takes care of the missing OutdoorAirNode.  It does not take care of the issue of the same node name on the inlet and outlet, so I have left that in.

Using DataLoopNode::ConnectionType::Inlet and Outlet means that when there is a severe error when paired heat pumps share matching inlet and outlet nodes:

```
   ** Severe  ** Node Connection Error, Node="OUTDOOR AIR INLET NODE", The same node appears as a non-parent Inlet node more than once.
   **   ~~~   ** Reference Object=HeatPump:PlantLoop:EIR:Cooling, Name=COOLING COIL
   **   ~~~   ** Reference Object=HeatPump:PlantLoop:EIR:Heating, Name=HEATING COIL
   ** Severe  ** Node Connection Error, Node="OUTDOOR AIR OUTLET NODE", The same node appears as a non-parent Outlet node more than once.
   **   ~~~   ** Reference Object=HeatPump:PlantLoop:EIR:Cooling, Name=COOLING COIL
   **   ~~~   ** Reference Object=HeatPump:PlantLoop:EIR:Heating, Name=HEATING COIL
```

Is this why node connection errors were handeled independently instead of relying on `BranchNodeConnections::CheckNodeConnections`?